indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -844,6 +858,48 @@ private void removeWorker(final Worker worker)
     }
   }
 
+  private void scheduleTasksCleanupForWorker(final String worker, final List<String> tasksToFail)
+  {
+    removedWorkerCleanups.put(
```

Comment:
If the worker is never seen again, then this map will not get rid of the values. Can we have a cleanup at the end of the listenable?